include/c2pa.hpp

```diff
@@ -115,4 +125,602 @@ namespace C2pa
         c2pa_release_string(result);
         return;
     }
-}
+
+    // IStream Class wrapper for CStream
+    class CppIStream : public CStream
+    {
+    public:
+        CStream *c_stream;
+
+        template <typename IStream>
+        CppIStream(IStream& istream) {
+            static_assert(std::is_base_of<std::istream, IStream>::value, "Stream must be derived from std::istream");
+            c_stream = c2pa_create_stream(reinterpret_cast<StreamContext*>(&istream), (ReadCallback)reader, (SeekCallback) seeker, (WriteCallback)writer, (FlushCallback)flusher);
+        }
```

Comment:
We also need to think about move constructors for the classes you've made. What happens to the underlying streams when you move the class? The default implemention may be enough, but we should make sure. Otherwise, we can delete them, like we deleted the copy operations. We can always implement move later!

Reply:
I'll review this in a later pass.